cyberark/conjur
# Porting the CE API docs into the rest of the CE docs site

Closes #126 

#### What does this pull request do?

This PR changes the default template of aglio when generating the CE API docs and instead creates a partial html file which can then be consumed by Jekyll in the `/docs/` site.

#### Where should the reviewer start?

The new template files for docs generation live under: ` apidocs/templates/`.  This will generate a new file under `/docs/_includes` called api.html.  The new files under `/docs/` are to create a new page in Jekyll for the apidocs, grab the partial, and insert it into the template like the other docs files.

#### How should this be manually tested?

To test this manually, a reviewer should:

1. Checkout the `api-docs-ui-revamp` possum branch into your local possum environment.
2. Follow the development steps listed here to build the possum docs sub-site:  https://github.com/conjurinc/possum/tree/api-docs-ui-revamp/docs#development
  * This will get the local docs site running at _http://localhost:4000_
  * At this point, the API Documentation link on the sidebar (http://localhost:4001/apidocs.html) should load a page with only the navigation and no page content.  This is expected.
3. Generate the API docs partial:
```
$ apidocs/build.sh
$ docker run --rm conjurinc/possum-apidocs > docs/_includes/api.html
```
  * This will run the api docs build script and then generate the partial in the expected location.
  *  The API Documentation page on your local environment should now have content!
 
#### Screenshots (if appropriate)

![image](https://user-images.githubusercontent.com/4164335/29048294-d586caf4-7b9d-11e7-84e1-0d0528eb80a4.png)

#### Link to build in Jenkins (if appropriate)

https://jenkins.conjur.net/job/possum/job/api-docs-ui-revamp/

#### Questions:
> Does this have automated Cucumber tests?
No

> Can we make a blog post, video, or animated GIF out of this?
No

> Is this explained in documentation?
This IS documentation! :)

> Does the knowledge base need an update?
No
